flexget/plugins/sites/alpharatio.py

```diff
@@ -0,0 +1,222 @@
+        :return:
+        """
+        cookies = self.get_login_cookie(username, password, force=force)
+
+        response = requests.get(url, params=params, cookies=cookies)
+
+        if self.base_url + 'login.php' in response.url:
+            if self.errors:
+                raise plugin.PluginError('AlphaRatio login cookie is invalid. Login page received?')
+            self.errors = True
+            # try again
+            response = self.get(url, params, username, password, force=True)
+        else:
+            self.errors = False
+
+        return response
+
+    def get_login_cookie(self, username, password, force=False):
+        """
+        Retrieves login cookie
+        :param username:
+        :param password:
+        :param force: if True, then retrieve a fresh cookie instead of looking in the DB
+        :return:
+        """
+        if not force:
+            with Session() as session:
+                saved_cookie = session.query(AlphaRatioCookie).filter(AlphaRatioCookie.username == username).first()
+                if saved_cookie and saved_cookie.expires and saved_cookie.expires >= datetime.datetime.now():
+                    log.debug('Found valid login cookie')
+                    return saved_cookie.cookie
+
+        url = self.base_url + 'login.php'
+        try:
+            log.debug('Attempting to retrieve AlphaRatio cookie')
+            response = requests.post(url, data={'username': username, 'password': password, 'login': 'Log in',
+                                                'keeplogged': '1'}, timeout=30)
+        except RequestException as e:
+            raise plugin.PluginError('AlphaRatio login failed: %s', e)
+
+        if 'Your username or password was incorrect.' in response.text:
+            raise plugin.PluginError('AlphaRatio login failed: Your username or password was incorrect.')
+
+        with Session() as session:
+            expires = None
+            for c in requests.cookies:
+                if c.name == 'session':
+                    expires = c.expires
+            if expires:
+                expires = datetime.datetime.fromtimestamp(expires)
+            log.debug('Saving or updating AlphaRatio cookie in db')
+            cookie = AlphaRatioCookie(username=username, cookie=dict(requests.cookies), expires=expires)
+            session.merge(cookie)
+            return cookie.cookie
+
+    @plugin.internet(log)
+    def search(self, task, entry, config):
+        """
+            Search for entries on AlphaRatio
+        """
+        params = {}
+
+        if 'category' in config:
+            categories = config['category'] if isinstance(config['category'], list) else [config['category']]
+            for category in categories:
+                params[CATEGORIES[category]] = 1
+
+        if 'scene' in config:
+            params['scene'] = int(config['scene'])
+
+        if 'order_desc' in config:
```

Comment:
This could be done in one line. You should probably set the default in schema instead of in the code.